codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/ValidationGenerator.java

```diff
@@ -398,14 +420,26 @@ private StringBuilder addRequiredCheck(
     ) {
       return requiredCheck;
     }
-    if (GoPointableIndex.of(model).isPointable(memberShape)) {
-      requiredCheck.append(
-        """
-        if ( %s == nil ) {
-            return fmt.Errorf(\"%s is required but has a nil value.\")
-        }
-        """.formatted(dataSource, dataSource)
-      );
+    final String nilCheck =
+      """
+      if ( %s == nil ) {
+          return fmt.Errorf(\"%s is required but has a nil value.\")
+      }
+      """;
+    if (
+      SmithyNameResolver.isShapeFromAWSSDK(targetShape) &&
+      memberShape.isOptional()
```

Comment:
You can use `AwsPointableIndex` but I'm also OK with this if that causes more problem. My concern is in a shape is Optional in aws-sdk but isn't pointable.